crates/workspace/src/workspace.rs

```diff
@@ -2655,6 +2656,20 @@ impl Workspace {
         }))
     }
 
+    pub fn join_pane_into_next(&mut self, pane: View<Pane>, cx: &mut ViewContext<Self>) {
+        let next_pane = self
+            .find_pane_in_direction(SplitDirection::Right, cx)
+            .or_else(|| self.find_pane_in_direction(SplitDirection::Left, cx));
+        let Some(next_pane) = next_pane else {
+            return;
+        };
+        let item_ids: Vec<EntityId> = pane.read(cx).items().map(|item| item.item_id()).collect();
```

Comment:
I did this because just iterating on `items()` would throw errors like:

```
error[E0502]: cannot borrow `*cx` as mutable because it is also borrowed as immutable
    --> crates/workspace/src/workspace.rs:2668:13
     |
2666 |         let pane_read_handle = pane.read(cx);
     |                                          -- immutable borrow occurs here
2667 |         for item in pane_read_handle.items() {
     |                     ------------------------ immutable borrow later used here
2668 |             self.move_item(pane.clone(), next_pane.clone(), item.item_id(), 0, cx);
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
```

I'm not extremely comfortable with Rust, so I welcome suggestions to remove this allocation if possible.

Reply:
I'm afraid there's no better way for this particular part of the code, but this is not a hot path, so why not.